Base/ctkAppLauncherEnvironment.h

```diff
@@ -0,0 +1,72 @@
+#ifndef __ctkAppLauncherEnvironment_h
+#define __ctkAppLauncherEnvironment_h
+
+// Qt includes
+#include <QObject>
+#include <QProcessEnvironment>
+
+class ctkAppLauncherEnvironmentPrivate;
+
+///
+/// \brief The ctkAppLauncherEnvironment class provides an interface for
+/// interacting with the AppLauncher environment variables.
+///
+/// It supports saving the environment and retrieving it. This allows
+/// the original launcher environment to be saved and later retrieved
+/// from the launched process.
+///
+/// Each time the environment is saved, if the environment variable `APPLAUNCHER_LEVEL`
+/// was set, its value `<level>` is incremented otherwise it is initialized to zero.
+/// Then all variables (*) are copied and named after `APPLAUNCHER_<level>_<varname>`.
+///
+/// (*) expect the one named `APPLAUNCHER_LEVEL` or starting with `APPLAUNCHER_<level>_`.
+///
+class ctkAppLauncherEnvironment : public QObject
+{
+  Q_OBJECT
+public:
+  typedef ctkAppLauncherEnvironment Self;
+  typedef QObject Superclass;
+
+  ctkAppLauncherEnvironment(QObject* parentObject = 0);
+  ~ctkAppLauncherEnvironment();
+
+  /// Returns the current applauncher level.
+  ///
+  /// This is the value of the `APPLAUNCHER_LEVEL`
+  /// environment variable.
+  ///
+  /// If the environment variable is not set or set to an invalid
```

Comment:
It is not clear what level=0 corresponds to. Is that the system environment?
Give a few examples: what is the level in a process that is started without applauncher, what is the level in an applauncher-started process; what is the value in a process started by that process.

Reply:
>  Is that the system environment?

Yes. It is the environment saved following the first invocation of any CTK launcher